tags.go

```diff
@@ -200,3 +206,48 @@ func (client *Client) DeleteTag(id ID) error {
 	err := client.Mutate(&m, v, WithName("TagDelete"))
 	return HandleErrors(err, m.Payload.Errors)
 }
+
+// ReconcileTags manages tags API operations for TaggableResourceInterface implementations
+//
+// Tags not in 'tagsWanted' will be deleted, new tags from 'tagsWanted' will be created. Reconciled tags are returned.
+func (client *Client) ReconcileTags(resourceType TaggableResourceInterface, tagsWanted []string) ([]Tag, error) {
+	var err error
+	var tagConnection *TagConnection
+	var assignedTags []Tag
+
+	tagConnection, err = resourceType.GetTags(client, nil)
+	if err != nil {
+		return assignedTags, err
+	}
+	if tagConnection == nil {
+		return assignedTags, fmt.Errorf("no tags found on %s with id '%s'", string(resourceType.ResourceType()), resourceType.ResourceId())
+	}
+
+	// delete tags found in resource but not listed in tagsWanted
+	for _, tag := range tagConnection.Nodes {
+		if !slices.Contains(tagsWanted, tag.Flatten()) {
+			if err := client.DeleteTag(tag.Id); err != nil {
+				return assignedTags, err
+			}
+		}
+	}
+
+	// format tags listed in Terraform config but not found in service
+	tagInput := map[string]string{}
+	for _, tag := range tagsWanted {
+		parts := strings.Split(tag, ":")
+		if len(parts) != 2 {
+			return assignedTags, fmt.Errorf("[%s] invalid tag, should be in format 'key:value' (only a single colon between the key and value, no spaces or special characters)", tag)
+		}
+		key := parts[0]
+		value := parts[1]
+		tagInput[key] = value
+	}
+	// assign tags listed in Terraform config but not found in service
+	assignedTags, err = client.AssignTags(string(resourceType.ResourceId()), tagInput)
```

Comment:
Should we add an additional method maybe `AssignTagsPro` that takes in the more structured `[]Tag` 🤔   Then the current `AssignTags` could defer to it and basically just be a function that converts `map[string]string` to `[]Tag` and calls `AssignTagsPro`